### PR TITLE
fix: support metadata in deferred calls execution

### PR DESCRIPTION
**Pull request by aludwiko**
References #1702

### CODE REVIEW

File: codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala [hunk 1]
```diff
@@ -173,6 +176,16 @@ object ComponentsSourceGenerator {
        |    return context.getComponentGrpcClient(serviceClass);
        |  }
        |
+       |  private <Req, Res> SingleResponseRequestBuilder<Req, Res> addHeaders(SingleResponseRequestBuilder<Req, Res> requestBuilder, Metadata metadata){
+       |    var updatedBuilder = requestBuilder;
+       |    for (Metadata.MetadataEntry entry: metadata){
+       |      if (entry.isText()) {
```
Comment (aludwiko) [inline codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala:182]: Actually, maybe for grpc based sdk, I should support binary headers as well?

File: samples/java-protobuf-valueentity-shopping-cart/src/test/java/com/example/shoppingcart/ShoppingCartActionImplTest.java [hunk 2]
```diff
@@ -47,7 +48,7 @@ public void tearDown() throws Exception {
   // tag::initialize[]
   @Test
   public void initializeCartTest() throws ExecutionException, InterruptedException, TimeoutException {
-    when(shoppingCartService.create(notNull()))
+    when(shoppingCartService.create().invoke(notNull()))
```
Comment (aludwiko) [inline samples/java-protobuf-valueentity-shopping-cart/src/test/java/com/example/shoppingcart/ShoppingCartActionImplTest.java:51]: Under the hood we have:
```
(Metadata metadata) -> addHeaders(((com.example.shoppingcart.ShoppingCartServiceClient) getGrpcClient(com.example.shoppingcart.ShoppingCartService.class)).create(), metadata).invoke(createCart)
```
thanks to `RETURNS_DEEP_STUBS` I was able to fix it in the java version, but with mockito scala I'm not sure if this is an option. 

Maybe we shouldn't encourage users to write unit tests for side effects. Personally I would go with IT tests for such cases.

**Review by octonato — APPROVED** [review]
LGTM

File: codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala (hunk 1 above)
Comment (johanandren) [reply codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala:182]: I think we can skip those for now.

**Review by octonato — APPROVED** [review]

**Pull request merged**